File: hw/vargas/u3/hw29Singleton/python/ec/edu/espe/singleton/utils/mongo_manager.py
from pymongo import MongoClient, errors
from bson.objectid import ObjectId
from datetime import datetime

class MongoManager:

    def __init__(self):
        uri = "mongodb+srv://Cesar:Cesar2006@cluster0.tgbv2qc.mongodb.net/"
        try:
            self.client = MongoClient(uri)
            self.database = self.client["toamedicalDB"]
        except errors.ConnectionFailure as e:
            print(f"Error de conexión a MongoDB: {e}")

    def insert(self, collection_name: str, document: dict):
        collection = self.database[collection_name]
        collection.insert_one(document)
        print(f"Documento insertado correctamente en: {collection_name}")

    def find(self, collection_name: str, filter_doc: dict) -> list:
        collection = self.database[collection_name]
        return list(collection.find(filter_doc))

    def get_email(self, collection_name: str, id_field_name: str, id_value) -> str:
        collection = self.database[collection_name]
        result = collection.find_one({id_field_name: id_value})
        
        if result and "email" in result:
            return result["email"]
        return None

    def get_info(self, collection_name: str, search_field: str, search_value, target_field: str) -> str:
        try:
            collection = self.database[collection_name]
            result = collection.find_one({search_field: search_value})
            
            if result and target_field in result:
                return str(result[target_field])
            return None
        except Exception as e:
            print(f"Error obteniendo info: {e}")
            return None

    def create_date_document(self, date_obj: datetime, hour: int, minute: int) -> dict:
        if date_obj is None:
            return None
        
        return {
            "date": date_obj.strftime("%Y-%m-%d"),
            "day": int(date_obj.strftime("%d")),
            "month": int(date_obj.strftime("%m")),
            "year": int(date_obj.strftime("%Y")),
            "hour": hour,
            "minute": minute,
            "time": f"{hour:02d}:{minute:02d}"
        }

    def close(self):
        if self.client:
            self.client.close()
            print("Conexión cerrada.")

File: hw/vargas/u3/hw29Singleton/javaScript/ec/edu/espe/singleton/utils/MongoManager.js
import { MongoClient, ObjectId } from 'mongodb';

export class MongoManager {
    constructor() {
        const uri = "mongodb+srv://Cesar:Cesar2006@cluster0.tgbv2qc.mongodb.net/";
        this.client = new MongoClient(uri);
        this.database = null;
    }

    async connect() {
        if (!this.database) {
            try {
                await this.client.connect();
                this.database = this.client.db("toamedicalDB");
            } catch (e) {
                console.error("Error conectando a MongoDB:", e);
            }
        }
    }

    async insert(collectionName, document) {
        await this.connect();
        const collection = this.database.collection(collectionName);
        await collection.insertOne(document);
        console.log(`Documento insertado correctamente en: ${collectionName}`);
    }

    async find(collectionName, filter = {}) {
        await this.connect();
        const collection = this.database.collection(collectionName);
        return await collection.find(filter).toArray();
    }

    async getEmail(collectionName, idFieldName, idValue) {
        await this.connect();
        const collection = this.database.collection(collectionName);
        const result = await collection.findOne({ [idFieldName]: idValue });
        
        return result?.email || null;
    }

    async getInfo(collectionName, searchField, searchValue, targetField) {
        await this.connect();
        try {
            const collection = this.database.collection(collectionName);
            const result = await collection.findOne({ [searchField]: searchValue });
            
            return result && result[targetField] ? String(result[targetField]) : null;
        } catch (e) {
            console.error(e);
            return null;
        }
    }

    createDateDocument(dateJs, hour, minute) {
        if (!dateJs) return null;

        const pad = (n) => String(n).padStart(2, '0');
        return {
            date: dateJs.toISOString().split('T')[0],
            day: dateJs.getDate(),
            month: dateJs.getMonth() + 1, 
            year: dateJs.getFullYear(),
            hour: hour,
            minute: minute,
            time: `${pad(hour)}:${pad(minute)}`
        };
    }

    async close() {
        if (this.client) {
            await this.client.close();
            console.log("Conexión cerrada.");
        }
    }
}